src/unstract/sdk/file_storage/helper.py

```
@@ -1,72 +1,58 @@
+            Sent as-is to the provider implementation.
 
         Returns:
             NA
         """
-        if provider == FileStorageProvider.GCS:
-            return FileStorageHelper.gcs_init(credentials)
-        elif provider == FileStorageProvider.Minio:
-            return FileStorageHelper.minio_init(credentials)
-        elif provider == FileStorageProvider.Local:
-            return FileStorageHelper.local_file_system_init()
 
-    @staticmethod
-    def gcs_init(credentials: dict[str, Any]) -> AbstractFileSystem:
-        """Initialises FileStorage backed up by GCS.
-
-        Args:
-            credentials (dict): credentials
-
-        Returns:
-            NA
-        """
         try:
-            gcs_creds = service_account.Credentials.from_service_account_file(
-                credentials[GCS.CREDS],
-                scopes=["https://www.googleapis.com/auth/cloud-platform"],
-            )
+            protocol = provider.value
+            if provider == FileStorageProvider.Local:
+                storage_config.update({"auto_mkdir": True})
+            elif provider in [FileStorageProvider.Minio]:
+                # Initialise using s3 for Minio
+                protocol = FileStorageProvider.S3.value
 
             fs = fsspec.filesystem(
-                protocol=FileStorageProvider.GCS.value,
-                project=credentials[GCS.PROJECT_ID],
-                token=gcs_creds,
+                protocol=protocol,
+                **storage_config,
             )
-            logger.debug(f"Connected to {FileStorageProvider.GCS.value} file system")
+            logger.debug(f"Connected to {provider.value} file system")
+        except KeyError as e:
+            logger.error(
+                f"Error in initialising {provider.value} "
+                f"file system because of missing config {e}",
+                stack_info=True,
+                exc_info=True,
+            )
+            raise FileStorageError(str(e))
```

@gaya3-zipstack this will display the `KeyError`'s message only in the caller which might not be useful

Is there any reason why we've logged this error with a traceback also? I think as a library we needn't do this and the logging of the traceback can remain within the client (Unstract) unless we suppress an exception anywhere
cc: @hari-kuriakose

---

Yeah... I have removed this now. Caller can log the details...